tensorflow/core/kernels/maxpooling_op_gpu.cu.cc

```diff
@@ -28,6 +28,21 @@ limitations under the License.
 
 namespace tensorflow {
 namespace {
+// Checks env var to determine whether maxpool functions should propagate NaN
+// values. NaNs are propagated unless TF_ENABLE_MAXPOOL_NANPROP=0.
+bool QueryMaxPoolNanPropIsEnabled() {
```

Comment:
Thinking about this. If we were to add this feature, we will want to test it both ways. We also cannot leave it as a global/static variable, as it won't change from test it test. 

An alternative is to leave it part of the OpKernelConstruction. So it's initialized the first as the graph constructs. Then we can easily give different settings per test.